src/Raven.Server/Documents/Handlers/AI/AiAgent/AiAgentHandler.cs

```diff
@@ -0,0 +1,415 @@
+                throw new ArgumentException($"AI Agent '{name}' doesn't exists");
+        }
+
+     
+
+        using var _ = ContextPool.AllocateOperationContext(out DocumentsOperationContext context);
+
+        var options = await context.ReadForMemoryAsync(RequestBodyStream(), "ai-agent", token.Token);
+        options.TryGet("Parameters", out BlittableJsonReaderObject parameters);
+
+        var r = await Talk(context, configuration, userPrompt, parameters, token: token);
+
+        string conversationId = null;
+        if (configuration.Persistence is not null)
+        {
+            MergedPutCommand putCmd = new(r.Dcoument, $"{configuration.Persistence.Collection}{Database.IdentityPartsSeparator}", null, Database);
+            await Database.TxMerger.Enqueue(putCmd);
+            conversationId = putCmd.PutResult.Id;
+        }
+
+        await using var writer = new AsyncBlittableJsonTextWriter(context, ResponseBodyStream());
+        writer.WriteStartObject();
+        writer.WritePropertyName("Response");
+        writer.WriteObject(r.Response.Result);
+        writer.WriteComma();
+        writer.WritePropertyName("Usage");
+        r.Usage.Write(writer);
+        writer.WriteComma();
+        writer.WritePropertyName("ChatId");
+        writer.WriteString(conversationId);
+        writer.WriteEndObject();
+    }
+
+    [RavenAction("/databases/*/ai/ai-agent/resume", "POST", AuthorizationStatus.ValidUser, EndpointType.Write)]
+    public async Task ResumeChat()
+    {
+        using var token = CreateHttpRequestBoundOperationToken();
+        var chatId = GetStringQueryString("chatId", required: true);
+        throw new NotImplementedException();
+    }
+
+    [RavenAction("/databases/*/ai/ai-agent/get", "GET", AuthorizationStatus.DatabaseAdmin)]
+    public async Task GetAiAgentConfiguration()
+    {
+        using var token = CreateHttpRequestBoundOperationToken();
+        var name = GetStringQueryString("agent", required: true);
+        AiAgentConfiguration configuration;
+        
+        using (ServerStore.Engine.ContextPool.AllocateOperationContext(out ClusterOperationContext ctx))
+        using (ctx.OpenReadTransaction())
+        using (var record = ServerStore.Cluster.ReadRawDatabaseRecord(ctx, DatabaseName))
+        {
+            if (record.TryGetAiAgent(name, out configuration) == false)
+                throw new ArgumentException($"AI Agent '{name}' doesn't exists");
+        }
+
+        using (ServerStore.ContextPool.AllocateOperationContext(out JsonOperationContext context))
+        await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBodyStream(), token.Token))
+        {
+            var obj = context.ReadObject(configuration.ToJson(), "get-ai-agent");
+            writer.WriteObject(obj);
+        }
+    }
+
+    [RavenAction("/databases/*/ai/ai-agent/get-all", "GET", AuthorizationStatus.DatabaseAdmin)]
+    public async Task GetAllAiAgents()
```

Comment:
Can we merge `/get` and `/get-all` ? So if you have an `agent` query string value, we'll return just that, otherwise, get all.

Reply:
sure